torchmetrics/utilities/exceptions.py

```diff
@@ -0,0 +1,19 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class MisconfigurationException(Exception):
```

Comment:
Can we rename this then? Typically UserError or RuntimeError is used for mis-use. Misconfiguration indicates that the initial config was wrong which does not always align with the usage here (I don't like it in PL, but it's hard to change there so maybe just correct it here?). So if we really want a specific error for TM, Can we make something like `TorchMetricsUserError` and maybe derive it from `UserError`?

Reply:
I like that !

Reply:
Done !